SSA/Experimental/IntrinsicAsymptotics.lean

```diff
@@ -15,10 +15,23 @@ open Goedel (toType)
   # Classes
 -/
 
+structure Signature (Ty : Type) where
+  sig     : List Ty
+  regSig  : List (Ctxt Ty × Ty)
+  outTy   : Ty
+
 class OpSignature (Op : Type) (Ty : outParam (Type)) where
-  sig : Op → List Ty
-  regSig : Op → List (Ctxt Ty × Ty)
-  outTy : Op → Ty
+  signature : Op → Signature Ty
```

Comment:
I guess we do it this way so we can talk about the `Signature` separate from the `Op` that produces it, right?

Reply:
What way does "this" refer to, before or after the change? I made the change so that `preserves_signature` down below in `DialectMorphism` can be stated as just one theorem, instead of having three theorems with mostly identical proofs.